test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertFrom-StringData.Tests.ps1

```diff
@@ -65,3 +65,50 @@ bazz = 2
 	$(ConvertFrom-StringData -StringData $sampleData).Values | Should -BeIn @("0","1","2")
     }
 }
+
+Describe "Delimiter parameter tests" -Tags "CI" {
+
+    $sampleData = @"
+
+a:b
+
+"@
+
+
+    It "Should return the data on the left side in the key" {
+        $actualValue = ConvertFrom-StringData -StringData 'a:b' -Delimiter ':'
+
+        $actualValue.Keys | Should -BeExactly "a"
+    }
+
+    It "Should return the data on the right side in the value" {
+        $actualValue = ConvertFrom-StringData -StringData 'a=b'
+
+        $actualValue.Values | Should -BeExactly "b"
+    }
+
+    It "Should not throw on given delimiter" {
+    {$sampleData | ConvertFrom-StringData -Delimiter ':' }| Should -Not -Throw
+    }
+
+
+    $TestCases = @(
+        @{ Delimiter = ':'; StringData = 'value:10'; ExpectedResult = @{ Values = 10 } }
+        @{ Delimiter = '-'; StringData = 'a-b' ; ExpectedResult = @{ Values = 'b' } }
+        @{ Delimiter = ','; StringData = 'c,d' ; ExpectedResult = @{ Values = 'd' } }
+    )
+
+    It 'is able to parse <StringData> with delimiter "<Delimiter>"' -TestCases $TestCases {
+        param($Delimiter, $StringData, $ExpectedResult)
+
+        $Result = ConvertFrom-StringData -StringData $StringData -Delimiter $Delimiter
+        foreach ($Name in $ExpectedResult.Keys) {
+            $Result.$Name | Should -Be $ExpectedResult.$Name
```

Comment:
Please add value check.

Reply:
This is actually checking both name _and_ value, at the same time. It just reads funny with that `$Name` Variable

Reply:
It looks a tricky. Could you please make the test more clear and explicitly check expected keys and value?

Reply:
Maybe something like this?
```powershell
$Name | Should -BeIn $Result.Keys
$Result.$Name | Should -Be $ExpectedResult.$Name
```

And then you could also rename `$Name` to `$Key` or something so it's slightly clearer.

Reply:
Done!